pkg/ignition/builder.go

```diff
@@ -99,10 +102,15 @@ func (b *ignitionBuilder) Generate() ([]byte, error) {
 		"/etc/NetworkManager/conf.d/clientid.conf",
 		0644, false,
 		[]byte("[connection]\nipv6.dhcp-duid=ll\nipv6.dhcp-iaid=mac")))
+
+	update_hostname := fmt.Sprintf(`
+	[[ "$DHCP6_FQDN_FQDN" =~ "." ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN 
+	[[ "$(< /proc/sys/kernel/hostname)" =~ (localhost|localhost.localdomain) ]] && hostnamectl set-hostname --transient %s`, b.hostname)
```

Comment:
nit: it's enough to do `=~ localhost`

Reply:
True, but they both incorrectly match all kinds of rubbish (like `test.definitely-not-a-localhost-domain.example.com` (the correct regex is `(^|\\.)localhost(\\.localdomain)?\\.?$`), so there's no point doing it wrong _and_ differently to https://github.com/openshift/machine-config-operator/blob/47436bdb7b8c49425d6813abca594485171e1221/templates/common/_base/files/usr-local-bin-mco-hostname.yaml#L17.